File: ruby/project.rb
class Project
  attr_accessor :projects

  def initialize
    @projects = {}
    @projects["project1"] = "Description of project1"
    @projects["project2"] = "Description of project2"
  end

  def list
    @projects.each do |key, value|
      print "#{key}, #{value}\n"
    end
  end

  def find(project_key)
    @projects.each do |key, value|
      if project_key == key
        print "#{key}, #{value}\n"
      end
    end
  end

  def create(name, desc)
    @projects[name] = desc
  end

  def update(name, desc)
    @projects[name] = desc
  end

  def destroy(name)
    @projects.delete(name)
  end
end

File: python/project.py
class Project(object):
  
  def __init__(self):
    self.projects = dict()
    self.projects["project1"] = "Description of project1"
    self.projects["project2"] = "Description of project2"

  def list(self):
    for key, value in sorted(self.projects.items()):
      print(key, value)

  def find(self, name):
    for key, value in self.projects.items():
      if key == name:
        print(key, value)

  def create(self, name, desc):
    self.projects[name] = desc

  def update(self, name, desc):
    self.projects[name] = desc

  def destroy(self, name):
    del self.projects[name]